cpp/src/arrow/compute/kernels/scalar_set_lookup_test.cc

```diff
@@ -253,11 +268,11 @@ TEST_F(TestIsInKernel, TimeDuration) {
               "[true, false, false, true, true]", /*skip_nulls=*/true);
   }
 
-  // Different units, invalid cast
-  ASSERT_RAISES(Invalid, IsIn(ArrayFromJSON(duration(TimeUnit::SECOND), "[0, 1, 2]"),
-                              ArrayFromJSON(duration(TimeUnit::MILLI), "[0, 2]")));
+  // Different units, cast value_set to values
+  CheckIsIn(ArrayFromJSON(duration(TimeUnit::SECOND), "[0, 2]"),
+            ArrayFromJSON(duration(TimeUnit::MILLI), "[1, 2, 2000]"), "[false, true]");
```

Comment:
More explicitly ensure that the cast actually happens (either 2 or 2000 could be the successful match...)
```suggestion
  CheckIsIn(ArrayFromJSON(duration(TimeUnit::SECOND), "[0, 1, 2]"),
            ArrayFromJSON(duration(TimeUnit::MILLI), "[1, 2, 2000]"), "[false, false, true]");
```

Reply:
Done